File: src/models/supplier.py
from sqlalchemy import Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from src.core.time import now_iso
from src.database.connection import Base


class Supplier(Base):
    __tablename__ = "suppliers"

    id: Mapped[int] = mapped_column(
        primary_key=True,
        autoincrement=True,
    )

    name: Mapped[str] = mapped_column(
        String(200),
        nullable=False,
    )

    document: Mapped[str | None] = mapped_column(
        String(50),
        nullable=True,
    )

    address: Mapped[str | None] = mapped_column(
        Text,
        nullable=True,
    )

    notes: Mapped[str | None] = mapped_column(
        Text,
        nullable=True,
    )

    is_active: Mapped[int] = mapped_column(
        Integer,
        nullable=False,
        default=1,
    )

    created_at: Mapped[str] = mapped_column(
        String(30),
        nullable=False,
        default=now_iso,
    )

    updated_at: Mapped[str] = mapped_column(
        String(30),
        nullable=False,
        default=now_iso,
        onupdate=now_iso,
    )

File: frontend/src/types/supplier.ts
export interface Supplier {
  id: number

  name: string
  document: string | null
  address: string | null
  notes: string | null

  is_active: boolean

  created_at: string
  updated_at: string
}

export interface SupplierCreatePayload {
  name: string
  document?: string | null
  address?: string | null
  notes?: string | null
}

export interface SupplierUpdatePayload {
  name?: string
  document?: string | null
  address?: string | null
  notes?: string | null
}

export interface SupplierDeactivatePayload {
  justification: string
}